File: Java/1614_Maximum_Nesting_Depth_of_the_Parentheses.java
class Solution {
    public int maxDepth(String s) {
        int result = 0;
        int temp = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '('){
                temp++;
                result = Math.max(result, temp);
            }
            if(s.charAt(i) == ')'){
                temp--;
            }
        }
        return result;
    }
}

File: Cpp/1614_Maximum_Nesting_Depth_of_the_Parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int result = 0;
        int temp = 0;
        for(char c : s){
            if(c == '('){
                temp++;
            }
            else if(c == ')'){
                temp--;
            }
            result = max(result, temp);
        }
        return result;
    }
};
